extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -124,8 +123,10 @@ void build(
 
         if (otelJdbcInstrumentationActive && capabilities.isPresent(OPENTELEMETRY_TRACER)) {
             // at least one datasource is using OpenTelemetry JDBC instrumentation,
-            // therefore we need to prepare OpenTelemetry data source wrapper
-            otelInstrumentationActiveProducer.produce(new OpenTelemetryJDBCInstrumentationBuildItem());
+            // therefore we register the OpenTelemetry data source wrapper bean
+            additionalBeans.produce(new AdditionalBeanBuildItem.Builder()
+                    .addBeanClass("io.quarkus.agroal.runtime.AgroalOpenTelemetryWrapper")
+                    .setDefaultScope(DotNames.SINGLETON).build());
```

Comment:
We now produce a bean that we will be able to inject into `DataSources` without making it mutable.

Reply:
Sure, that's better.